parler_tts/modeling_parler_tts.py

```diff
@@ -1323,16 +1315,42 @@ def forward(
         else:
             raise ValueError("You have to specify either decoder_input_ids or decoder_inputs_embeds")
 
-        # past_key_values_length
-        past_key_values_length = past_key_values[0][0].shape[2] if past_key_values is not None else 0
-
         if inputs_embeds is None:
             inputs_embeds = sum([self.embed_tokens[codebook](input[:, codebook]) for codebook in range(num_codebooks)])
 
         # if prompt_hidden_states, fuse to inputs_embeds and update input shape
         if prompt_hidden_states is not None:
```

Comment:
We should keep track of prompt sequence length to add it to `past_key_values_length` afterwards:
```suggestion
        prepended_sequence_length = 0
        # if prompt_hidden_states, fuse to inputs_embeds and update input shape
        if prompt_hidden_states is not None:
            prepended_sequence_length = prompt_hidden_states.shape[-2]
```

Reply:
done in [41edc2a](https://github.com/huggingface/parler-tts/pull/89/commits/41edc2aeca462e7de48fe8f11f9896343a8f7ae2)